compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -2187,11 +2190,63 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
                 .unwrap_or("'lt".to_string())
         };
 
-        let add_lt_sugg = generics
-            .params
-            .first()
-            .and_then(|param| param.def_id.as_local())
-            .map(|def_id| (self.tcx.def_span(def_id).shrink_to_lo(), format!("{}, ", new_lt)));
+        let mut add_lt_suggs: Vec<Option<_>> = vec![];
+        if is_synthetic {
+            if let Some(ast_generics) = ast_generics {
+                if ast_generics.params.iter().any(|p| {
+                    matches!(
+                        p.kind,
+                        hir::GenericParamKind::Lifetime { kind: hir::LifetimeParamKind::Explicit }
+                    )
+                })
+                && let Some(def_id) = ast_generics.params.first().and_then(|param| Some(param.def_id)) {
```

Comment:
I find this `if` condition very hard to read. Maybe store the values in appropriately named variables first?

Reply:
Introduced `named_lifetime_exist` variable.